src/ImageSharp/Formats/Jpeg/Port/Components/FrameComponent.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="FrameComponent.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Formats.Jpeg.Port.Components
+{
+    using System;
+
+    using ImageSharp.Memory;
+
+    /// <summary>
+    /// Represents a single frame component
+    /// </summary>
+    internal struct FrameComponent : IDisposable
+    {
+        /// <summary>
+        /// Gets or sets the component Id
+        /// </summary>
+        public byte Id;
+
+        /// <summary>
+        /// TODO: What does pred stand for?
+        /// </summary>
+        public int Pred;
+
+        /// <summary>
+        /// Gets or sets the horizontal sampling factor.
+        /// </summary>
+        public int HorizontalFactor;
+
+        /// <summary>
+        /// Gets or sets the vertical sampling factor.
+        /// </summary>
+        public int VerticalFactor;
+
+        /// <summary>
+        /// Gets or sets the identifier
+        /// </summary>
+        public byte QuantizationIdentifier;
+
+        /// <summary>
+        /// Gets or sets the block data
+        /// </summary>
+        public Buffer<short> BlockData;
```

Comment:
Or .. maybe it's better to allocate it as a private member `Buffer<short> blockData` to be `ArrayPool`-friendly, but we should expose it only through a `public Span<Block8x8Int> BlockData { get; }` property utilizing `NonPortableCast`.